rust/src/cli/service.rs

```diff
@@ -99,3 +111,86 @@ fn relocate_file(file_path: &Path) -> Result<(), CliError> {
     );
     Ok(())
 }
+
+fn pin_iface_name(cfg_dir: &Path) -> Result<(), CliError> {
+    let file_path = cfg_dir.join(PIN_STATE_FILENAME);
+    let mut fd = std::fs::File::open(&file_path)?;
+    let mut content = String::new();
+    fd.read_to_string(&mut content)?;
+    let pin_state: NetworkState = serde_yaml::from_str(&content)?;
```

Comment:
Although, it's actually a bit more efficient to let serde do the reading from the file directly (not that this matters) like this:

```
    let mut r = std::fs::File::open(&file_path).map(std::io::BufReader::new);
    let pin_state: NetworkState = serde_yaml::from_reader(&mut r)?;
```

to avoid allocating an intermediate string.